Schedule has no Teams field

I started writing a fix for PagerDuty/terraform-provider-pagerduty#261, but found that this library also needs support for the Teams property of Schedules.

```
pagerduty/resource_pagerduty_schedule.go:147:11: schedule.Teams undefined (type *pagerduty.Schedule has no field or method Teams)
```

Raising the issue here to see if we can support this property end-to-end.

Cheers.

--- a/pagerduty/schedule.go
+++ b/pagerduty/schedule.go
@@ -33,6 +33,7 @@ type Schedule struct {
 	TimeZone             string                       `json:"time_zone,omitempty"`
 	Type                 string                       `json:"type,omitempty"`
 	Users                []*UserReference             `json:"users,omitempty"`
+	Teams                []*TeamReference             `json:"teams,omitempty"`
 }
 
 // SubSchedule represents a sub-schedule of a schedule.

